### PR TITLE
Fixes #246

### LINKED ISSUE #246: PanelNavigationService breaks irrecoverably if a panel is moved to a region that it does not belong to and the user tries to close it via the ribbon.
Open a panel in either the Left or Right region. Drag the panel to a different region then its default. Try to close the panel via its button in the ribbon. The culprit in the service:

image

Other actions related to drag and drop of panels may also break the system. The DockingManager does not seem to interact directly with the defined custom groups.

Other odd behaviors:

- Move one panel from left to right. Then try to open a Left panel. No longer possible.
- Move panel from left to right. Open a right panel, the new panel does not open in the tabbed group.

At the moment it is not certain that the region manager workflow is at all necessary and used correctly.

Investigate the correct devexpress way described in https://docs.devexpress.com/WPF/15540/Controls-and-Libraries/Layout-Management/Dock-Windows/Miscellaneous/Managing-Dock-Panels-in-Code

### PR DESCRIPTION
**Pull request by lxatrhea**
- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
- Fixed bugs in issue #246 
- Fixed unit tests

Scenario's described in issue #246 should not happen anymore.

### CODE REVIEW

File: CDP4Composition/Navigation/Interfaces/IRegionCollectionSearcher.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    using Microsoft.Practices.Prism.Regions;
+
+    /// <summary>
+    /// Helper class for searching views in a <see cref="IRegionCollection"/>
+    /// </summary>
+    public interface IRegionCollectionSearcher
+    {
+        /// <summary>
+        /// Search a <see cref="IRegionCollection"/> for <see cref="IRegion"/>s that contain a specific <see cref="IPanelView"/>
+        /// /// </summary>
```
Comment (alexatrhea) [inline CDP4Composition/Navigation/Interfaces/IRegionCollectionSearcher.cs]: redundant `///`

File: CDP4Composition/Navigation/Interfaces/IRegionCollectionSearcher.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        /// <param name="regionManagerRegions">The <see cref="IRegionCollection"/> to search</param>
+        /// <param name="view">The <see cref="IPanelView"/> to search for</param>
+        /// <returns></returns>
```
Comment (alexatrhea) [inline CDP4Composition/Navigation/Interfaces/IRegionCollectionSearcher.cs]: Return missing

File: CDP4Composition/Navigation/Interfaces/IRegionCollectionSearcher.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        IEnumerable<IRegion> RegionsForView(IRegionCollection regionManagerRegions, IPanelView view);
```
Comment (alexatrhea) [inline CDP4Composition/Navigation/Interfaces/IRegionCollectionSearcher.cs]: The name of this method is a bit vague. I would suggest something like `GetRegionsByView()`.

File: CDP4Composition/Navigation/RegionCollectionSearcher.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    using System.Collections.Generic;
+    using System.ComponentModel.Composition;
+    using System.Linq;
+
+    using Microsoft.Practices.Prism.Regions;
+
+    [Export(typeof(IRegionCollectionSearcher))]
+    public class RegionCollectionSearcher : IRegionCollectionSearcher
+    {
+        public IEnumerable<IRegion> RegionsForView(IRegionCollection regionManagerRegions, IPanelView view)
```
Comment (alexatrhea) [inline CDP4Composition/Navigation/RegionCollectionSearcher.cs]: Docs and name change from interface.

File: CDP4Composition/Navigation/RegionCollectionSearcher.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+namespace CDP4Composition
+{
+    using System.Collections.Generic;
+    using System.ComponentModel.Composition;
+    using System.Linq;
+
+    using Microsoft.Practices.Prism.Regions;
+
+    [Export(typeof(IRegionCollectionSearcher))]
+    public class RegionCollectionSearcher : IRegionCollectionSearcher
```
Comment (alexatrhea) [inline CDP4Composition/Navigation/RegionCollectionSearcher.cs]: Docs.

Comment (samatrhea) [inline CDP4Composition/Navigation/RegionCollectionSearcher.cs]: missing some docs here

**Pull request merged**